srcpkgs/gsoap/template

```diff
@@ -1,36 +1,44 @@
 # Template file for 'gsoap'
 pkgname=gsoap
-version=2.8.91
-revision=3
+version=2.8.124
+revision=1
 build_style=gnu-configure
 configure_args="--enable-ipv6"
-hostmakedepends="unzip flex autoconf"
+hostmakedepends="unzip flex autoconf automake libtool"
 makedepends="openssl-devel zlib-devel"
 short_desc="Toolkit for SOAP and REST Web Services and XML-Based Applications"
 maintainer="Enno Boland <gottox@voidlinux.org>"
-license="gSOAP-1.3b, GPL-2.0-only"
-homepage="http://www.cs.fsu.edu/~engelen/soap.html"
+license="GPL-2.0-only"
+homepage="https://www.genivia.com/dev.html"
 distfiles="${SOURCEFORGE_SITE}/gsoap2/gsoap_${version}.zip"
-checksum=2ca21b5ab2b40d899566c559f71a1d51ed7eac3f00160ab78060027a3ee6488f
+checksum=4b798780989338f665ef8e171bbcc422a271004d62d5852666d5eeca33a6a636
 disable_parallel_build=yes
-
 CFLAGS="-fcommon"
 
 if [ "$CROSS_BUILD" ]; then
 	hostmakedepends+=" gsoap"
 fi
 
+post_patch() {
+	if [ "${XBPS_TARGET_LIBC}" = "musl" ]; then
+		cp -f "${FILESDIR}/musl-fixes.patch" "${wrksrc}"
+		cd "$wrksrc"
+		msg_normal "$pkgver: patching for musl\n"
+		patch -s -Np1 <musl-fixes.patch 2>/dev/null
+	fi
+}
```

Comment:
This is very ugly, does the patch not work on glibc?

Reply:
The patch does not work with glibc.

Reply:
can this be merged, even if ugly?

Reply:
did you try looking into why this doesn't patch work on glibc?

Reply:
The `gethostbyname_r` and `strerror_r` are different in gnu and posix. The original code tries to check this with `!defined(_GNU_SOURCE)`, but maybe something else is turning on `gnu_source`?

Reply:
could you test if https://github.com/void-linux/void-packages/pull/41844/files#r1123699454 or something similar could fix this? (at least the strerror part)

Reply:
that seems to work, thanks. I'll try smth similar for the other part.